feat(website): getting started revamp (#1285)
cdk8s-team/cdk8s: docs/get-started/java.md

Next, we’ll initialize a project to create the directory structure and install the necessary dependencies using the [init](https://cdk8s.io/docs/latest/cli/init/) command. 

1. In a terminal window, run the following command in an empty directory: 
```console
cdk8s init java-app
```

1. In your preferred code editor, open the `Main.java` file. 
```java
package com.mycompany.app;

import software.constructs.Construct;

import org.cdk8s.App;
import org.cdk8s.Chart;
import org.cdk8s.ChartProps;

public class Main extends Chart 
{

    public Main(final Construct scope, final String id) {
        this(scope, id, null);
    }

    public Main(final Construct scope, final String id, final ChartProps options) {
        super(scope, id, options);

        // define resources here
    }

    public static void main(String[] args) {
        final App app = new App();
        new Main(app, "java");
        app.synth();
    }
}
```

This sample shows the basic structure of a cdk8s application with the essential libraries: **constructs** and **cdk8s**. These libraries supply the foundational classes and methods required for working with cdk8s. It includes the following components:

>    * A custom `MyChart` class inherits from the [Chart](https://cdk8s.io/docs/latest/reference/cdk8s/java/#chart) base class provided by the cdk8s library, serving as a representation of the Kubernetes resources to be generated and managed.

Reviewer: ```suggestion
   * A custom `Main` class inherits from the [Chart](https://cdk8s.io/docs/latest/reference/cdk8s/java/#chart) base class provided by the cdk8s library, serving as a representation of the Kubernetes resources to be generated and managed.
```

>    * The `MyChart` constructor method is responsible for initializing the base class ([Chart](https://cdk8s.io/docs/latest/reference/cdk8s/java/#chart)) and specifying the Kubernetes resources. This method is invoked when creating an instance of the class.

Reviewer: ```suggestion
   * The `Main` constructor method is responsible for initializing the base class ([Chart](https://cdk8s.io/docs/latest/reference/cdk8s/java/#chart)) and specifying the Kubernetes resources. This method is invoked when creating an instance of the class.
```

>    * An instance of the [App](https://cdk8s.io/docs/latest/reference/cdk8s/java/#app) class signifies the primary entry point of the cdk8s application and oversees the application's lifecycle and resources.

Reviewer: ```suggestion
   * An instance of the [App](https://cdk8s.io/docs/latest/reference/cdk8s/java/#app) class is created in the `main` function, which acts the primary entry point of the cdk8s application and oversees the application's lifecycle and resources.
```

>    * An instance of the `MyChart` class is created by passing the app instance and a string identifier, "java", as arguments. This action generates and registers the Kubernetes resources defined in the `MyChart` class within the application. Note that in this example, the "java" string identifier is autogenerated based on the current directory name.

Reviewer: ```suggestion
   * An instance of the `Main` class is created by passing the app instance and a string identifier, "java", as arguments. This action generates and registers the Kubernetes resources defined in the `Main` class within the application. Note that in this example, the "java" string identifier is autogenerated based on the current directory name.
```

>    * The `synth` method is called on the `app` instance, which produces the required Kubernetes YAML manifest files based on the defined resources. Note that in this example, we haven't defined any resources within the `MyChart` constructor, so running the "cdk8s synth" command in the CLI would generate a blank Kubernetes manifest.

Reviewer: ```suggestion
   * The `synth` method is called on the `app` instance, which produces the required Kubernetes YAML manifest files based on the defined resources. Note that in this example, we haven't defined any resources within the `Main` constructor, so running the "cdk8s synth" command in the CLI would generate a blank Kubernetes manifest.
```

# Getting started with cdk8s for Java
In this guide, we'll walk you through the process of building a basic Kubernetes application using cdk8s for Java.

## Prerequisites
1. To install the cdk8s CLI, you need the [Node Package Manager (npm)](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) installed on your local machine. 
2. cdk8s for Java supports Java Development Kit (JDK) versions 8+. Check your Java version:
```bash
java --version
```

## Set up an environment
### Initial setup
To get setup, follow these general steps:

1. To begin, you'll need Maven 3.6.3+ installed. If you're not familiar with this setup, use these instructions to [setup Apache Maven](https://maven.apache.org/install.html).

### Install the CLI
To initialize a cdk8s project and auto-generate Kubernetes manifests based on our code, we need the cdk8s CLI:

> 1. Run the following command to install the CLI using [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm). For more installation methods, see [Install the cdk8s CLI](cli/installation.md).

Reviewer: ```suggestion
1. Run the following command to install the CLI using [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm). For more installation methods, see [Install the cdk8s CLI](./../cli/installation.md).
```

```console
npm install -g cdk8s-cli
```

### Create a project

> Next, we’ll initialize a project to create the directory structure and install the necessary dependencies using the [init](https://cdk8s.io/docs/latest/cli/init/) command. 

Reviewer: ```suggestion
Next, we’ll initialize a project to create the directory structure and install the necessary dependencies using the [init](./../cli/init.md) command. 
```
